apps/docusaurus/docs/sdks/ts-sdk/type-safe-contract.md

```diff
@@ -0,0 +1,49 @@
+# Surf: Type Safety for your Move Contracts
+
+## What is Surf
+
+Surf is TypeScript library built on top of Aptos TypeScript SDK and wallet adaptor that provides type safety for your Move contracts. Type safety is important because it allows you to catch errors at compile time rather than at runtime, and usually IDE like VS Code shows warning if you try to access a field that doesn't exist in the contract or if you pass the wrong type to a function.
+
+## Usage
+
+You first need to download the ABI of the Move contract you want to interact with. 
+
+```bash gen_abi.sh
+#! /bin/bash
+
+NETWORK=testnet
+# replace it with your contract address
+CONTRACT_ADDRESS=0x12345
+# replace it with your package name, you can find it in Move.toml
+PACKAGE_NAME=fungible_asset_launchpad
+
+echo "export const ABI = $(curl https://fullnode.$NETWORK.aptoslabs.com/v1/accounts/$CONTRACT_ADDRESS/module/$PACKAGE_NAME | sed -n 's/.*"abi":\({.*}\).*}$/\1/p') as const" > src/utils/abi.ts
+```
+
+Then you can use surf client as a replacement of the Aptos client when interact with Move contracts, both in frontend and node script. For non-contract related operations like managing account, querying indexer API, you still need to use Aptos client. 
+
+```ts src/utils/aptos.ts
+import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
+import { ABI } from "./abi";
+
+const config = new AptosConfig({
+  network: Network.TESTNET,
+});
+// You used to use aptosClient to interact with the chain
+export const aptosClient = new Aptos(config);
+// Now you use surfClient that wraps aptosClient to interact with Move contracts
+export const surfClient = createSurfClient(aptosClient).useABI(ABI);
```

Comment:
lets add some more examples like how to interact with entry/view functions? can even introduce a full example on getting the `0x1::account` module ABI, and then submitting a transfer transaction and calling a view function

Reply:
I lean to this page only serves as an intro, user should go to surf github to learn usage in detail. Surf has a very good readme, even a video demo.

https://github.com/ThalaLabs/surf